File: 2° BackEnd/app/main/subModels/game.py
from django.db import models
from main.subModels.theme import Theme
from main.subModels.studio import Studio
from main.subModels.status import Status
from main.subModels.language import Language
from main.subModels.gametype import GameType
from main.subModels.producer import Producer
from main.subModels.plataform import Plataform
from main.subModels.perspective import Perspective

from django.contrib.postgres.indexes import GinIndex

class Game(models.Model):
    release = models.DateField()
    playtime = models.FloatField(default = 0.0)
    source = models.CharField(default="gameAPI", max_length = 50, blank=False, null=False)
    premiede = models.CharField(default="unknow", max_length = 30, blank=False, null=False)
    gameImage = models.CharField(max_length = 300, blank=False, null=False)
    description = models.CharField(max_length = 500, blank=False, null=False)
    name = models.CharField(max_length = 100)

    theme =  models.ManyToManyField(Theme)
    studio = models.ManyToManyField(Studio)
    producer = models.ManyToManyField(Producer)
    status = models.ManyToManyField(Status)
    language = models.ManyToManyField(Language)
    gameType = models.ManyToManyField(GameType) #Fps, Moba and Etc
    plataform = models.ManyToManyField(Plataform)
    perspective = models.ManyToManyField(Perspective)

    class Meta:
        indexes = [
            GinIndex(name='NewGinIndex', fields=['name'])
        ]

    def __str__(self):
        return f"\n| Id: {self.id}\n| gameImage: {self.gameImage}\n| Name: {self.name}"

File: 3° FrontEnd/src/app/shared/models/Game.ts

export class Game {
    private _id?: number;
    private _score?: number;
    private _name?: string;
    private _playtime?: number;
    private _release?: string;
    private _premiede?: string;
    private _gameImage?: string;
    private _description?: string;
    private _theme?: any;
    private _studio?: any;
    private _producer?: any;
    private _status?: any;
    private _language?: any;
    private _gameType?: any;
    private _plataform?: any;
    private _perspective?: any;

    constructor(id: number, name: string);
    constructor(...myarray: any[]){
        if (myarray.length === 2) {
            this._id = myarray[0]
            this.name = myarray[1]
            return;
        }
    }

    static simpleRefract(game: Game) {
        const result = {
            id: game!.id,
            score: game!.score,
            name: game!.name,
            playtime: game!.playtime,
            release: game!.release,
            premiede: game!.premiede,
            gameImage: game!.gameImage,
            description: game!.description,
            theme: game!.theme,
            studio: game!.studio,
            producer: game!.producer,
            status: game!.status,
            language: game!.language,
            gameType: game!.gameType,
            plataform: game!.plataform,
            perspective: game!.perspective
        }
        return result;
    }

    get id(): number | undefined{
        return this._id;
    }

    set id(id: number | undefined){
        this._id = id;
    }

    get score() : number | undefined {
        return this._score;
    }

    set score(score : number | undefined) {
        this._score = score;
    }

    get name() : string | undefined {
        return this._name;
    }

    set name(name : string | undefined) {
        this._name = name;
    }

    get playtime() : number | undefined {
        return this._playtime;
    }

    set playtime(playtime: number | undefined) {
        this._playtime = playtime;
    }

    get release() : string | undefined {
        return this._release;
    }

    set release(release: string | undefined) {
        this._release = release;
    }

    get premiede() : string | undefined {
        return this._premiede;
    }

    set premiede(premiede: string | undefined) {
        this._premiede = premiede;
    }

    get gameImage() : string | undefined {
        return this._gameImage;
    }

    set gameImage(gameImage: string | undefined) {
        this._gameImage = gameImage;
    }

    get description() : string | undefined {
        return this._description;
    }

    set description(description: string | undefined) {
        this._description = description;
    }

    get theme() : any | undefined {
        return this._theme;
    }

    set theme(theme: any | undefined) {
        this._theme = theme;
    }

    get studio() : any | undefined {
        return this._studio;
    }

    set studio(studio: any | undefined) {
        this._studio = studio;
    }

    get producer() : any | undefined {
        return this._producer;
    }

    set producer(producer: any | undefined) {
        this._producer = producer;
    }

    get status() : any | undefined {
        return this._status;
    }

    set status(status: any | undefined) {
        this._status = status;
    }

    get language() : any | undefined {
        return this._language;
    }

    set language(language: any | undefined) {
        this._language = language;
    }

    get gameType() : any | undefined {
        return this._gameType;
    }

    set gameType(gameType: any | undefined) {
        this._gameType = gameType;
    }

    get plataform() : any | undefined {
        return this._plataform;
    }

    set plataform(plataform: any | undefined) {
        this._plataform = plataform;
    }

    get perspective() : any | undefined {
        return this._perspective;
    }

    set perspective(perspective: any | undefined) {
        this._perspective = perspective;
    }
}